### ISSUE TITLE
first parameter or limit causes `Unknown field 'fieldName' on type 'Query'.`

### ISSUE THREAD

**Issue by heldrida** [gatsbyjs/gatsby#13765]
## Description

When passing a variable to the Gatsby-source-graphql graphql function causes the error `Unknown field 'fieldName' on type 'Query'.`; The `fieldName` is the property in gatsby-config.js

A tested query in Graphiql works fine, but not in `gatsby-source-graphql`

My stack is gratsbyjs + wordpress graphql wp plugin

### Steps to reproduce

`import React from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'gatsby'
import Home from 'components/Home'
import _has from 'lodash/has'

const Index = ({ data }) => <Home posts={_has(data, 'wpcms.posts') && data.wpcms.posts} />

Index.propTypes = {
  data: PropTypes.object.isRequired
}

export default Index

export const query = graphql`
  query {
    wpcms {
      posts (first: 3) {
        pageInfo {
          hasNextPage
          endCursor
        }
        edges {
          cursor
          node {
            id
            title
            slug
            date
            excerpt
            author {
              id
              name
            }
            featuredImage {
              id
              sourceUrl
              slug
            }
          }
        }
      }
    }
  }
`
`

### Expected result

If I execute the query as a HTTP Post request using Axios, works fine! And the same query in Graphiql works fine! For example:

`query {
  posts (first: 10 after: "YXJyYXljb25uZWN0aW9uOjg4MTg=") {
    pageInfo {
      hasNextPage
      endCursor
    }
    edges {
      cursor
      node {
        id
        title
        slug
        date
        excerpt
        author {
          id
          name
        }
        featuredImage {
          id
          sourceUrl
          slug
        }
      }
    }
  }
}
`

### Actual result

Even if I put the `limit` or `first` in the fieldName does not work (a different error is thrown `- Unknown argument 'limit'.`, obviously not expecting it to work in the `fieldName`, but tried to see)

The error in the original query above is:

- Unknown field 'wpcms' on type 'Query'.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

Did not work in 2.5.7, no output

**Comment by heldrida** [comment]
Here is a paste.ee of the `src/pages/index.js` file ( https://paste.ee/p/83AUF#BunbvcT6To9QOQC8hFqKeKciVyzTwOP2 )

Tested in the graphiql without the fieldName in the query, works absolutely fine! and through a HTTP Post.

**Comment by heldrida** [comment]
Ok, so I changed my gatsby-node.js, that has a createPage that is not the src/pages/index.js but src/component/Blog/Post. Both with the same query! Ran the gatsby clean and dev command with success and no errors thrown. The hell?! The createPage is not even related to the index page!

`  query {
    wpcms {
      posts (first: 3) {
        pageInfo {
          hasNextPage
          endCursor
        }
        edges {
          cursor
          node {
            id
            title
            slug
            date
            excerpt
            author {
              id
              name
            }
            featuredImage {
              id
              sourceUrl
              slug
            }
          }
        }
      }
    }
  }`

To keep my sanity, removed the parameter first in `posts (first: 3) { ... }`, for the gatsby-node.js file. Ran the gatsby clean and dev command and works fine and no errors thrown!

There's something very weird going on here, it's random! I know I sound crazy but; damn, I was chatting about this for days in the discord gatsby chat room, done, the tests I've exposed above; How is that? I'll see if the error shows up again at some point. But there's something going on here. The change I did was literally replicating the query in the -node.js file...
